test/Airgrab.test.ts

```diff
@@ -0,0 +1,351 @@
+        // Send the message to fractal api and get the proof
+        const res = await fetch(url);
+        fractalCredentials = await res.json();
+      } catch (err) {
+        console.log(err);
+        throw new Error('Error fetching credentials from Fractal API');
+      }
+
+      // Claim params
+      address = fractalCredentials.address;
+      proof = fractalCredentials.proof;
+      validUntil = fractalCredentials.validUntil;
+      approvedAt = fractalCredentials.approvedAt;
+      fractalId = fractalCredentials.fractalId;
+    });
+
+    it('Should fail with invalid KYC', async function () {
+      const userSigner = await ethers.getImpersonatedSigner(testUser);
+      const invalidFractalProof = '0xdeaddeaddead';
+
+      // Claim airgrab
+      await expect(
+        airgrab
+          .connect(userSigner)
+          .claim(
+            claimAmount,
+            address,
+            process.env.MERKLE_PROOF!.split(','),
+            invalidFractalProof,
+            validUntil,
+            approvedAt,
+            fractalId,
+          ),
+      ).to.be.revertedWith('Airgrab: Invalid KYC');
+    });
+
+    it('Should fail with invalid merkle proof', async function () {
+      const userSigner = await ethers.getImpersonatedSigner(testUser);
+      // Claim airgrab
+      await expect(
+        airgrab
+          .connect(userSigner)
+          .claim(
+            claimAmount,
+            address,
+            process.env.MERKLE_PROOF!.split(',').slice(0, -1),
+            proof,
+            validUntil,
+            approvedAt,
+            fractalId,
+          ),
+      ).to.be.revertedWith('Airgrab: not in tree');
+    });
+
+    it('Should allow only the claimer', async function () {
+      const alice = ((await ethers.getSigners()) as HardhatEthersSigner[])[0]!;
+
+      // Claim airgrab
+      await expect(
+        airgrab
+          .connect(alice)
+          .claim(
+            claimAmount,
+            address,
+            process.env.MERKLE_PROOF!.split(','),
+            proof,
+            validUntil,
+            approvedAt,
+            fractalId,
+          ),
+      ).to.be.revertedWith('Airgrab: Invalid KYC');
+    });
+
+    it('Should revert when the KYC is expired', async function () {
+      const userSigner = await ethers.getImpersonatedSigner(testUser);
+
+      // Time travel to the future to expire the KYC
+      const now = await helpers.time.latest();
+      await helpers.time.increase(BigInt(validUntil - now + 1));
+
+      // Claim airgrab
+      await expect(
+        airgrab
+          .connect(userSigner)
+          .claim(
+            claimAmount,
+            address,
+            process.env.MERKLE_PROOF!.split(','),
+            proof,
+            validUntil,
+            approvedAt,
+            fractalId,
+          ),
+      ).to.be.revertedWith('Airgrab: KYC no longer valid');
+    });
+
+    it('Should be successfull using a KYCed & eligible account', async function () {
```

Comment:
I thought we wanted to have an account/test for each blocked country?

Reply:
Chatted about this offline, won't do due to the way Fractal sandbox works (TL;DR: users seem to be manually approved/rejected, so doing this won't add much value)